src/js/components/DataTable/Body.js

```diff
@@ -45,34 +45,30 @@ const Row = memo(
         size={size}
         active={active}
         aria-disabled={(onClickRow && isDisabled) || undefined}
-        onClick={
-          onClickRow && !isDisabled
-            ? (event) => {
-                // extract from React's synthetic event pool
-                event.persist();
-                const adjustedEvent = event;
-                adjustedEvent.datum = datum;
-                adjustedEvent.index = index;
-                onClickRow(adjustedEvent);
-              }
-            : undefined
-        }
+        onClick={(event) => {
+          if (onClickRow && !isDisabled) {
+            if (typeof onClickRow === 'function') {
+              // extract from React's synthetic event pool
+              event.persist();
+              const adjustedEvent = event;
+              adjustedEvent.datum = datum;
+              adjustedEvent.index = index;
+              onClickRow(adjustedEvent);
+            }
+            if (onClickRow === 'onSelect') {
```

Comment:
```suggestion
            else if (onClickRow === 'onSelect') {
```
liking need to fix formatting locally w linter.

Reply:
done